src/tests/ftest/launch.py

```diff
@@ -20,164 +20,465 @@
-    test_cmd = avocado + ignore_errors + output_options
-    if use_tags:
-        test_cmd += category
-    test_cmd += params + ' -- ' + _file
+    tag_filter = ""
+    test_list = []
+    for tag in tags:
+        if ".py" in tag:
+            # Assume '.py' indicates a test and just add it to the list
+            test_list.append(tag)
+        elif tag == "all":
+            # Use an empty filter to find all tests
+            tag_filter = " "
+            break
+        else:
+            # Otherwise it is assumed that this is a tag
+            tag_filter += " --filter-by-tags={}".format(tag)
 
-    start_time = int(time.time())
-    print("Running: " + test_cmd + "\n\n")
-    subprocess.call(test_cmd, shell=True)
-    end_time = int(time.time())
-    print("Total test run-time in seconds: {}".format(end_time - start_time))
-    # move the job to a path that will be recognizable
-    try:
-        os.makedirs(os.path.join("avocado", os.path.dirname(_file)))
-    except OSError as excepn:
-        if excepn.errno != errno.EEXIST:
-            raise
-    os.rename(os.path.realpath(os.path.join("avocado", "job-results",
-                                            "latest")),
-              os.path.join("avocado", _file))
-    os.remove(os.path.join("avocado", "job-results", "latest"))
-    os.rmdir(os.path.join("avocado", "job-results"))
+    if not tag_filter == "":
+        # Find all the tests that match the tags
+        command = "avocado list --paginator off{} ./ | " \
+                  "grep '^INSTRUMENTED' | awk '{{print $2}}' | " \
+                  "awk -F ':' '{{print $1}}' | uniq".format(tag_filter)
+        test_list.extend(get_output(command).splitlines())
 
+    return tag_filter, test_list
 
-if __name__ == "__main__":
 
-    # not perfect param checking but good enough for now
-    sparse = False
-    addnl_tests = []
-    if len(sys.argv) == 2:
-        if sys.argv[1] == '-h' or sys.argv[1] == '--help':
-            printhelp()
-        test_request = sys.argv[1]
-    elif len(sys.argv) >= 3:
-        if sys.argv[1] == '-s':
-            sparse = True
-            test_request = sys.argv[2]
-        if sparse:
-            remaining_args = 3
+def get_test_files(test_list, machine_mapping, debug):
+    """Get a list of the test scripts to run and their yaml files.
+
+    Args:
+        test_list (list): list of test scripts to run
+        machine_mapping (dict): the mapping of machine replacement keys to
+            their replacement values
+        debug (bool): whether or not to dispaly debug messages
+
+    Returns:
+        list: a list of dictionaries of each test script and yaml file
+
+    """
+    test_files = [{"py": test, "yaml": None} for test in test_list]
+    home_tmp = os.path.join(os.path.expanduser("~"), "tmp")
+    get_output("mkdir -p {}".format(home_tmp))
+    for test_file in test_files:
+        base, _ = os.path.splitext(test_file["py"])
+        file = os.path.basename(base)
```

Comment:
(pylint-redefined-builtin) Redefining built-in 'file'

Reply:
Fixed